network/src/noise/handshake.rs

```diff
@@ -132,13 +135,41 @@ pub struct NoiseUpgrader {
 
 impl NoiseUpgrader {
     /// Create a new NoiseConfig with the provided keypair and authentication mode.
-    pub fn new(key: x25519::PrivateKey, auth_mode: HandshakeAuthMode) -> Self {
+    pub fn new(peer_id: PeerId, key: x25519::PrivateKey, auth_mode: HandshakeAuthMode) -> Self {
         Self {
+            peer_id,
             noise_config: noise::NoiseConfig::new(key),
             auth_mode,
         }
     }
 
+    /// Convert a remote pubkey into a network `PeerId`.
+    fn identity_pubkey_to_peer_id(&self, remote_pubkey: x25519::PublicKey) -> io::Result<PeerId> {
+        // if mutual authentication, try to find peer with this identity pubkey
+        if let Some(trusted_peers) = self.auth_mode.trusted_peers() {
+            trusted_peers
+                .read()
+                .map_err(|_| {
+                    io::Error::new(
+                        io::ErrorKind::Other,
+                        "noise: unable to read trusted_peers lock",
+                    )
+                })?
+                .iter()
```

Comment:
maybe assign the read.map_err? to an intermediate variable, this is a pretty long chain

Reply:
(I removed that function in the second commit)